pkg/api/v1alpha1/machine_types.go

```diff
@@ -0,0 +1,15 @@
+package v1alpha1
+
+type OSFamily string
+
+const (
+	Ubuntu       OSFamily = "ubuntu"
+	Bottlerocket OSFamily = "bottlerocket"
+	Redhat       OSFamily = "redhat"
```

Comment:
If we refactor this, does this mean we support `Redhat` for VSphere too potentially?

Reply:
We are exploring an idea of supporting a yum based distro. Till we have something solid, should we also put this behind a feature flag?

Reply:
I see, I think it's okay to not need a feature flag for this because these aren't values we display in the CLI, and we specifically validate against Ubuntu and Bottlerocket here: https://github.com/aws/eks-anywhere/blob/39d53e7d1e49252d27d14367db9048e770f4128b/pkg/providers/vsphere/validator.go#L90

Reply:
I would suggest creating a different type for cloud stack OS's
That allows us to maintain two separate "collections" of values (is more semantic that anything else)

Reply:
I'm not sure I agree - that'll lead to redundancy. Fundamentally, the os family concept is exactly the same between the providers, we just need to have proper validation for each one to make sure only the supported ones are allowed, right? 

If you feel strongly otherwise, I'm happy to disagree and commit though

Reply:
Putting this convo on pause until [we decide](https://github.com/aws/eks-anywhere/pull/1076#discussion_r801148963) if we even need `OSFmaily` in the cloud stack machines :)